pyrules/rules.py

```diff
@@ -1,93 +1,161 @@
     """
-    condition = lambda self, context: False
-    action = lambda self, context: None 
+    def __init__(self, condition=None, action=None):
+        self._condition = condition
+        self._action = action
+
+    def condition(self, context):
+        """
+        Condition for executing this rule.
+
+        Override in subclasses if necessary. Should return boolean value that
+        determines if rule is used.
+        """
+        return self._condition(self, context)
+
+    def action(self, context):
+        """
+        Action for executing this rule.
+
+        Override in subclasses if necessary. Should return dictionary with
+        results that will be added to context.
+        """
+        return self._action(self, context)
+
     def should_trigger(self, context):
         return self.condition(context)
+
     def perform(self, context):
         result = self.action(context) 
         context.update(result)
         return result
-    
-class TableRuleset(Rule):
+
+
+class TableRule(Rule):
     """
-    a table ruleset created from a list of dict objects of the following format
+    A table rule is created from a list of dict objects of the following format:
     
     [ 
       {
-        'if' : ['condition1', ...],
+        'if' : {'logic': '1 | 2', 'conditions': ['foo', {'bar': 10}]},
         'then' : ['action1', ...],
         'target' : ['target1', ...]
       }, 
       ...
     ]
     
-    Each rule is only executed if all conditions are met. In conditions and actions, use context. 
-    to reference variables. targets implicitly reference context. (target 'xy' means 'context.xy').
-    The result of the nth 'then' action is stored in the nth context.variable as defined in target.
+    Each rule is only executed if all conditions are met. In actions, use
+    'context.' to reference variables. Targets and conditions implicitly reference
+    'context.' (target 'xy' means 'context.xy'). Logic can be omitted, which
+    would imply "&" operation for all conditions. Condition can be a dictionary or
+    a single value, so 'value' is equivalent to {'value': True}
+
+    The result of the nth 'then' action is stored in the nth 'context.variable'
+    as defined in target.
     """
-    def __init__(self, rules, translations=None):
+    def __init__(self, rules, name=None):
         self.rules = rules or {}
-        if translations:
-            translator = Translator(translations)
-            for rule in self.rules:
-                for key in rule.keys():
-                    rule[key] = [translator.replace(item) for item in rule[key]]
-    def should_trigger(self, context):
-        return True
+        if name:
+            self.name = name
+        self._current_ruleid = None
+
     def perform(self, context):
         count = 0
         for rule in self.rules:
-            if all([eval(condition, context.as_dict) for condition in rule['if']]):
+            evaluator = LogicEvaluator(
```

Comment:
this being the same for every call of a rule, why not move it to `_from_data`? Combined with my previous comment on C objects parsing operators, this would give us parse-time error messages if logic or conditions cannot be parsed.